src-cpp/rdkafkacpp_int.h

```diff
@@ -964,7 +964,7 @@ class HandleImpl : virtual public Handle {
                                    std::string &errstr) {
           char errbuf[512];
           ErrorCode err;
-          const char **extensions_copy = new const char *[extensions.size()];
```

Comment:
This was actually a bug, new:ed memory should be freed with delete, not free().
Since this memory is not related to librdkafka and only lives for this call I think we should use the original new here, and change the free to delete;

Reply:
Well, since it's passed to the C function it makes sense to be malloc-ed, I guess, and free was called anyway, so in that case delete has to be called.

Reply:
The allocator doesn't matter here since the memory will never be transfered to the application or librdkafka, it is just some temporary buffer, so let's use new/delete.